src/main/kotlin/com/idan/pokemon_hub/init/PokemonInitializer.kt

```diff
@@ -0,0 +1,95 @@
+package com.idan.pokemon_hub.init
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.jacksonTypeRef
+import com.idan.pokemon_hub.model.Pokemon
+import com.idan.pokemon_hub.model.PokemonImage
+import com.idan.pokemon_hub.model.PokemonType
+import com.idan.pokemon_hub.repository.PokemonRepository
+import jakarta.annotation.PostConstruct
+import org.slf4j.LoggerFactory
+import org.springframework.context.annotation.Profile
+import org.springframework.core.io.ClassPathResource
+import org.springframework.stereotype.Component
+import java.io.IOException
+import java.net.URI
+
+@Component
+@Profile("!test") // Don't run in the "test" profile
```

Comment:
There's no need for all these comments.
You should avoid comments all the time.
People forget to update them, and they become outdated really fast.

If you feel the need to add the comments, then maybe you need to refactor the code so it's more readable by itself. 

In this case the code fragment is so simple, the comment doesn't add any value.
Please remove it.